lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/restart/DFSEvaluatorLogAppendReaderWriter.java

```diff
@@ -65,6 +66,11 @@ public synchronized void writeToEvaluatorLog(final String formattedEntry) throws
     }
   }
 
+  @Override
+  public Iterable<String> readFromEvaluatorLog() throws IOException {
```

Comment:
So, the iterable opens the file? Not when it creates the Iterator? We need to be explicit here to make sure that resources are always released. One way would be to read the file when generating the iterator. Then, we can also close the file at the same time.